src/main/java/com/avail/interpreter/levelTwo/operation/L2_GET_INVALID_MESSAGE_RESULT_FUNCTION.java

```diff
@@ -47,6 +47,7 @@
 import org.objectweb.asm.MethodVisitor;
 
 import java.util.Set;
+import java.util.function.Consumer;
```

Comment:
Any reason why this class doesn't have `hasSideEffect` set to `true`? Seems like the same rationale ought to apply as for  `L2_GET_CURRENT_FUNCTION` and `L2_GET_IMPLICIT_OBSERVE_FUNCTION`.

Reply:
There's nothing interesting about this `import`, of course, there just weren't many places where GitHub would let me attach a comment about this file.

Reply:
I added @ReadsHiddenVariable(GLOBAL_STATE.class) to the class, and to its twin, L2_GET_INVALID_MESSAGE_SEND_FUNCTION.  It has no side-effects, so the instruction can be dropped if the code needing the function disappears (i.e., reading the global variable is only necessary if you need the value).  I verified that the primitives that set these global functions all use Flags.WritesToHiddenGlobalState, which keeps these operations from commuting.

L2_GET_CURRENT_FUNCTION is a different situation.  It isn't reading a global, just Interpreter.currentFunction, so it's already annotated with @ReadsHiddenVariable(LATEST_RETURN_VALUE.class), which keeps it from commuting with instructions that destroy that field.  The calling convention change will eventually remove that whole instruction.